contracts/peripheral/grants/VeBalGrant.sol

```diff
@@ -0,0 +1,352 @@
+        WETH.approve(address(BALANCER_VAULT), type(uint256).max);
+        BAL.approve(address(BALANCER_VAULT), type(uint256).max);
+    }
+
+    /* ----------------------------------------------------------------
+       Modifiers 
+    ---------------------------------------------------------------- */
+
+    /**
+     * @notice Modifier that allows only Project or Balancer can trigger a function
+     */
+    modifier onlyAuth() {
+        require(msg.sender == project || msg.sender == balancer, "!auth");
+        _;
+    }
+
+    /**
+     * @notice Modifier that allows only Project to trigger a function
+     */
+    modifier onlyProject() {
+        require(msg.sender == project, "!project");
+        _;
+    }
+
+    /**
+     * @notice Modifier that allows only Balancer to trigger a function
+     */
+    modifier onlyBalancer() {
+        require(msg.sender == balancer, "!balancer");
+        _;
+    }
+
+    /**
+     * @notice Modifier that only allows something to be called when the contract is active
+     */
+    modifier whileActive() {
+        require(active, "!active");
+        _;
+    }
+
+    /**
+     * @notice Modifier that only allows something to be called when the contract is inactive
+     */
+    modifier whileInactive() {
+        require(!active, "active");
+        _;
+    }
+
+    /* ----------------------------------------------------------------
+       Shared Functions
+    ---------------------------------------------------------------- */
+
+    /**
+     * @notice Releases veBAL lock
+     * grant must be inactive in order for this to be called
+     */
+    function release() external onlyAuth whileInactive {
+        votingEscrow.withdraw();
+        hasLock = false;
+    }
+
+    /**
+     * @notice exits BPT position in return for WETH and BAL
+     * grant must be inactive in order for this to be called
+     */
+    function redeem() external onlyAuth whileInactive {
+        _exitBalEthPool();
+    }
+
+    /**
+     * @notice Sends WETH and BAL to project and balancer
+     * Project gets WETH up to the amount they contributed in the initial lock
+     * Balancer gets all BAL and any remaining WETH
+     * grant must be inactive in order for this to be called
+     */
+    function withdrawBalances() external onlyAuth whileInactive {
+        uint256 wethForProjectBalance = AuraMath.min(ethContributed, WETH.balanceOf(address(this)));
+        WETH.transfer(project, wethForProjectBalance);
+        WETH.transfer(balancer, WETH.balanceOf(address(this)));
+        BAL.transfer(balancer, BAL.balanceOf(address(this)));
+        ethContributed = 0;
+    }
+
+    /**
+     * @notice Allows Balancer or Project to vote for a gauge
+     * @param gauge      gauge that will be voted for
+     * @param weight     vote weight
+     */
+    function voteGaugeWeight(address gauge, uint256 weight) external {
+        active ? require(msg.sender == project, "!caller") : require(msg.sender == balancer, "!caller");
```

Comment:
Nit: maybe it's just me but general I would expect ternaries to be used when the return value is being consumed. Might be neater to write this as an if statement or do the ternary `msg.sender == active ? project : balance`. Same for `execute`

Reply:
Was just looking to replace that awful if-else. Adopted your version